File: Random Stuff For Alex/TileData.cpp
#include "TileData.h"

#include <stdlib.h>
#include <stdint.h>
#include <stdbool.h>

TileData::TileData(uint32_t id, uint32_t levelColor, bool solid) {
	this->id = id;
	this->levelColor = levelColor;
	this->solid = solid;
	tiles = (TileData*) malloc(sizeof(TileData) * 256);
	tiles[id] = *this;
}

uint32_t TileData::get_id(void) {
	return this->id;
	return this->id;
}

uint32_t TileData::get_level_color(void) {
	return this->levelColor;
}

uint32_t TileData::is_solid(void) {
	return this->solid;
}

File: Random Stuff For Alex/TileData.java
package com.ludusamo.LD28.levels.tiles;

public class TileData {
	public static TileData[] tiles = new TileData[256];
	public static TileData VOID;
	public static TileData STONE;
	public static TileData GRASS;

	private int id;
	private int levelColor;
	private boolean isSolid;

	public TileData(int id, int levelColor, boolean isSolid) {
		this.id = id;
		this.levelColor = levelColor;
		this.isSolid = isSolid;

		tiles[id] = this;
	}

	public int getId() {
		return id;
	}

	public int getLevelColor() {
		return levelColor;
	}

	public boolean isSolid() {
		return isSolid;
	}

}
